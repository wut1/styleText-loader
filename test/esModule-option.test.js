import path from 'path';

import {
  compile,
  getCompiler,
  getEntryByInjectType,
  getErrors,
  getWarnings,
  runInJsDom,
} from './helpers/index';

describe('参数"esModule"', () => {
    const entry = './simple.js';
    it(`没有指定的时候应该可用`, async () => {
      const compiler = getCompiler(entry);
      const stats = await compile(compiler);

      runInJsDom('main.bundle.js', compiler, stats, (dom) => {
        expect(dom.serialize()).toMatchSnapshot('DOM');
      });

      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });

    it(`指定为"true"的时候应该可用`, async () => {
     
      const compiler = getCompiler(entry, {esModule: true });
      const stats = await compile(compiler);

      runInJsDom('main.bundle.js', compiler, stats, (dom) => {
        expect(dom.serialize()).toMatchSnapshot('DOM');
      });

      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });

    it(`指定为"true" 并且设置CSS modules时候应该可用`, async () => {
  
      const compiler = getCompiler(
        entry,
        { },
        {
          module: {
            rules: [
              {
                test: /\.css$/i,
                use: [
                  {
                    loader: path.resolve(__dirname, '../src/cjs.js'),
                    options: {esModule: true },
                  },
                   { loader: 'css-loader', options: { modules: true } },
                ],
              },
            ],
          },
        }
      );
      const stats = await compile(compiler);

      runInJsDom('main.bundle.js', compiler, stats, (dom) => {
        expect(dom.serialize()).toMatchSnapshot('DOM');
      });

      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });

    it(`指定为"true" 并且未设置CSS modules的时候应该可用`, async () => {
      const compiler = getCompiler(
        entry,
        { },
        {
          module: {
            rules: [
              {
                test: /\.css$/i,
                use: [
                  {
                    loader: path.resolve(__dirname, '../src/cjs.js'),
                    options: { esModule: true },
                  },
                   {
                        loader: 'css-loader',
                        options: { modules: false },
                      },
                ],
              },
            ],
          },
        }
      );
      const stats = await compile(compiler);

      runInJsDom('main.bundle.js', compiler, stats, (dom) => {
        expect(dom.serialize()).toMatchSnapshot('DOM');
      });

      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });

    it(`设置为"false"时应该可用`, async () => {
    
      const compiler = getCompiler(entry, {esModule: false });
      const stats = await compile(compiler);

      runInJsDom('main.bundle.js', compiler, stats, (dom) => {
        expect(dom.serialize()).toMatchSnapshot('DOM');
      });

      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });

    it(`设置为"false"时并且设置CSS modules应该可用`, async () => {
    
      const compiler = getCompiler(
        entry,
        { },
        {
          module: {
            rules: [
              {
                test: /\.css$/i,
                use: [
                  {
                    loader: path.resolve(__dirname, '../src/cjs.js'),
                    options: {esModule: false },
                  },
                   { loader: 'css-loader', options: { modules: true } },
                ],
              },
            ],
          },
        }
      );
      const stats = await compile(compiler);

      runInJsDom('main.bundle.js', compiler, stats, (dom) => {
        expect(dom.serialize()).toMatchSnapshot('DOM');
      });

      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });

    it(`设置为"false"时并且未设置CSS modules应该可用`, async () => {
   
      const compiler = getCompiler(
        entry,
        { },
        {
          module: {
            rules: [
              {
                test: /\.css$/i,
                use: [
                  {
                    loader: path.resolve(__dirname, '../src/cjs.js'),
                    options: {esModule: false },
                  },
                  {
                        loader: 'css-loader',
                        options: { modules: false },
                      },
                ],
              },
            ],
          },
        }
      );
      const stats = await compile(compiler);

      runInJsDom('main.bundle.js', compiler, stats, (dom) => {
        expect(dom.serialize()).toMatchSnapshot('DOM');
      });

      expect(getWarnings(stats)).toMatchSnapshot('warnings');
      expect(getErrors(stats)).toMatchSnapshot('errors');
    });
  
});
