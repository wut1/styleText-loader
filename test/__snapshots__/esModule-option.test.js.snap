// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`参数"esModule" 指定为"true" 并且未设置CSS modules的时候应该可用: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`参数"esModule" 指定为"true" 并且未设置CSS modules的时候应该可用: errors 1`] = `Array []`;

exports[`参数"esModule" 指定为"true" 并且未设置CSS modules的时候应该可用: warnings 1`] = `Array []`;

exports[`参数"esModule" 指定为"true" 并且设置CSS modules时候应该可用: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`参数"esModule" 指定为"true" 并且设置CSS modules时候应该可用: errors 1`] = `Array []`;

exports[`参数"esModule" 指定为"true" 并且设置CSS modules时候应该可用: warnings 1`] = `Array []`;

exports[`参数"esModule" 指定为"true"的时候应该可用: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`参数"esModule" 指定为"true"的时候应该可用: errors 1`] = `Array []`;

exports[`参数"esModule" 指定为"true"的时候应该可用: warnings 1`] = `Array []`;

exports[`参数"esModule" 没有指定的时候应该可用: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`参数"esModule" 没有指定的时候应该可用: errors 1`] = `Array []`;

exports[`参数"esModule" 没有指定的时候应该可用: warnings 1`] = `Array []`;

exports[`参数"esModule" 设置为"false"时并且未设置CSS modules应该可用: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`参数"esModule" 设置为"false"时并且未设置CSS modules应该可用: errors 1`] = `Array []`;

exports[`参数"esModule" 设置为"false"时并且未设置CSS modules应该可用: warnings 1`] = `Array []`;

exports[`参数"esModule" 设置为"false"时并且设置CSS modules应该可用: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`参数"esModule" 设置为"false"时并且设置CSS modules应该可用: errors 1`] = `Array []`;

exports[`参数"esModule" 设置为"false"时并且设置CSS modules应该可用: warnings 1`] = `Array []`;

exports[`参数"esModule" 设置为"false"时应该可用: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`参数"esModule" 设置为"false"时应该可用: errors 1`] = `Array []`;

exports[`参数"esModule" 设置为"false"时应该可用: warnings 1`] = `Array []`;
